crates/stages/src/stages/sender_recovery.rs

```diff
@@ -108,16 +109,21 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
                 reth_db::DatabaseError,
             >,
                            rlp_buf: &mut Vec<u8>|
-             -> Result<(u64, H160), Box<StageError>> {
-                let (tx_id, transaction) = entry.map_err(|e| Box::new(e.into()))?;
+             -> Result<(u64, H160), Box<SenderRecoveryStageError>> {
+                let (tx_id, transaction) =
+                    entry.map_err(|e| Box::new(SenderRecoveryStageError::StageError(e.into())))?;
                 let tx_id = tx_id.key().expect("key to be formated");
 
                 let tx = transaction.value().expect("value to be formated");
                 tx.transaction.encode_without_signature(rlp_buf);
 
-                let sender = tx.signature.recover_signer(keccak256(rlp_buf)).ok_or(
-                    StageError::from(SenderRecoveryStageError::SenderRecovery { tx: tx_id }),
-                )?;
+                let sender =
+                    match tx.signature.recover_signer(keccak256(rlp_buf)) {
+                        Some(sender) => Ok(sender),
+                        None => Err(SenderRecoveryStageError::FailedRecovery(
+                            FailedSenderRecoveryError { tx: tx_id },
+                        )),
+                    }?;
```

Comment:
let's prefer `ok_or`
```suggestion
                let sender = tx.signature.recover_signer(keccak256(rlp_buf)).ok_or(
                    SenderRecoveryStageError::FailedRecovery(FailedSenderRecoveryError {
                        tx: tx_id,
                    }),
                )?;
```